sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,94 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=5, normalize=True):
+    """Top k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that
+    multilabel classification case isn't covered here.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels.
+
+    y_score : array-like of shape (n_samples, n_classes)
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values as returned by :term:`predict_proba` or
+        :term:`decision_function`.
+
+    k : int, default=5
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, default=True
+        If `True`, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        The best performance is 1 with `normalize == True` and the number of
+        samples with `normalize == False`.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
```

Comment:
Hmm, thinking about this more, I think we should typically be permissive in case of ties, and still count it has a hit when we can.

Maybe it would be reasonably easy to implement with the `order` parameter of `argsort`?

Reply:
Would require `y_score` input to be a numpy array specifying at least one field and adding an optional `order` parameter to `top_k_accuracy_score` that then will be pass to `argsort`.

I just hope that `y_score = check_array(y_score, ensure_min_features=3)` doesn't remove fields from the array.

Should I give a try?

Reply:
I think we would need to create the structured array ourselves, after `check_array` is called

Reply:
After investigating we are actually using [`np.argmax`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.argmax.html#numpy-argmax) in classifiers' `predict` method (cf. [in linear_model](https://github.com/scikit-learn/scikit-learn/blob/95d4f0841/sklearn/linear_model/_base.py#L279)).

`np.argmax` documentation states:
> In case of multiple occurrences of the maximum values, the indices corresponding to the first occurrence are returned.

Should we anyway do something about ties here?

Reply:
Let's leave it as-is and wait for others opinion then

Reply:
I would ordinarily err on the side of underestimating performance, rather than being "permissive in case of ties". I think tie breaking in other metrics, such as label ranking metrics like coverage_error, similarly prefers to understate performance.

The current approach provides the benefit that it is consistent with accuracy given scikit-learn classifier conventions. I don't mind it. A conservative tie-breaking approach would also be reasonable.